### PR TITLE
[WIP] rule parser

### PR DESCRIPTION
**Pull request by Ozonid**

### CODE REVIEW

File: server/rules.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          case '$lte':
+            return rule.value >= sensorValue
+          case '$lt':
+            return rule.value > sensorValue
+          case '$gt':
+            return rule.value < sensorValue
+        }
+      })
+      if (ruleResults[mode](Boolean)) {
+        // send alert somehow
```
Comment (Ozonid) [inline server/rules.js]: or maybe just return the value and do the alert from the device itself

File: server/rules.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          case '$in':
+            return rule.value.indexOf(sensorValue) !== -1
+          case '$nin':
+            return rule.value.indexOf(sensorValue) === -1
+          case '$eq':
+            return rule.value === sensorValue
+          case '$ne':
+            return rule.value !== sensorValue
+          case '$gte':
+            return sensorValue >= rule.value
```
Comment (tothandras) [inline server/rules.js]: `rule.value <= sensorValue`

File: server/rules.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  checkValues (sensors) {
+    this.rules.forEach((entry) => {
+      const mode = entry.conditions.mode === 'all' ? 'every' : 'some'
+      const ruleResults = entry.conditions.rules.map((rule) => {
+        const sensorValue = sensors.get(rule.channel.name).latestValue
+        switch (rule.operator) {
+          case '$in':
+            return rule.value.indexOf(sensorValue) !== -1
+          case '$nin':
+            return rule.value.indexOf(sensorValue) === -1
```
Comment (tothandras) [inline server/rules.js]: `<0`

File: server/rules.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  checkValues (sensors) {
+    this.rules.forEach((entry) => {
+      const mode = entry.conditions.mode === 'all' ? 'every' : 'some'
+      const ruleResults = entry.conditions.rules.map((rule) => {
+        const sensorValue = sensors.get(rule.channel.name).latestValue
+        switch (rule.operator) {
+          case '$in':
+            return rule.value.indexOf(sensorValue) !== -1
```
Comment (tothandras) [inline server/rules.js]: `> -1`

File: server/rules.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            return sensorValue >= rule.value
+          case '$lte':
+            return rule.value >= sensorValue
+          case '$lt':
+            return rule.value > sensorValue
+          case '$gt':
+            return rule.value < sensorValue
+        }
+      })
+      if (ruleResults[mode](Boolean)) {
```
Comment (tothandras) [inline server/rules.js]: I see :)

**Pull request closed without merging**